src/ros2cs/rosidl_generator_cs/cmake/rosidl_generator_cs_generate_interfaces.cmake

```diff
@@ -223,15 +235,116 @@ foreach(_generated_msg_c_ts_file ${_generated_msg_c_ts_files})
     ${PROJECT_NAME}__rosidl_generator_c
   )
 
-  set(ros2_distro "$ENV{ROS_DISTRO}")
+  rosidl_target_interfaces(${_target_name}
+    ${PROJECT_NAME} rosidl_typesupport_c
+  )
 
-  if(ros2_distro STREQUAL "humble" OR ros2_distro STREQUAL "rolling")
-    rosidl_get_typesupport_target(c_typesupport_target "${PROJECT_NAME}" "rosidl_typesupport_c")
```

Comment:
Use new API for humble and up (https://docs.ros.org/en/humble/Releases/Release-Humble-Hawksbill.html#deprecation-of-rosidl-target-interfaces)